### ISSUE TITLE
Fix IP Address detection-followup

### ISSUE THREAD

**Issue by SteveEbey73742** [UnitedRPMs/minidlna#1]
I am forced to restart the service, after logging into the desktop to get the network module to find the adapter. I have changed config, and added waits in the service startup file, to no avail. Can the code be changed to wait for a valid IP, before the service loads and exits.

**Comment by paulcarroty** [comment]
@SteveEbey73742 you can create a fake network interface with your IP and setup it inside minidlna config. Also use the systemd unit with network depends:

```
[Unit]
Wants=network-online.target
After=network-online.target
```

Feel free to reopen this issue if meet this problem again.

**Comment by SteveEbey73742** [comment]
how do i fake a network? and where in minidlna.conf do i add / change to the fake network? i have the service waits in already. What has worked for me, as a workaround, is to completely ignore the network interface, and just let minidlna run on the only interface it finds, instead of using the named interface.

**Comment by paulcarroty** [comment]
Let me understand what you mean under `a valid IP`? You need a different IP for minidlna?

**Comment by SteveEbey73742** [comment]
Let me be clear, I have a static ip on my nic.  If I tell minidlna the nic name in the config, I have to force the service to restart, to let it listen.  It is not seeing the network interface name, so I quit telling it the name, just comment the option out in the conf file, and it runs on ALL available interfaces.  Since only one nic is in use in my computer, it finds the nic, but if i try to use the nic name enp2s0 it tells me it can not find it.

here is current conf that works.
```
# network interfaces to serve, comma delimited
# network_interface=enp2s0

```
if I uncomment the name line, the service fails to start, and I have to systemctl restart minidlna after I login.

**Comment by kuboosoft** [comment]
@SteveEbey73742 really? it runs in ALL available interfaces? 
https://unix.stackexchange.com/a/311844
https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/

**Comment by sergiomb2** [comment]
please also address this issue to RPMFusion

**Comment by SteveEbey73742** [comment]
@kuboosoft Yes, if you have wlan and wired or have multiple nics in your servier, it uses them all to listen on. the original implementation, as started by netgear, was to listen to all the ports in a media ready router.  specifying a single interface to listen on, allows users to regulate access from the router.  Some NAS routers from netgear still use a variant of this, and call it MedisShare.

**Comment by kuboosoft** [comment]
@SteveEbey73742 by default the /etc/minidlna.conf is provided virgin (all commented);  maybe you need [configurate locally (per user)](https://help.ubuntu.com/community/MiniDLNA) I am not a expert with minidlna but maybe it works for you... I am updating minidlna to version [1.1.6](https://sourceforge.net/p/minidlna/git/ci/d1cd7c7a4a4bcceb11f7e285324a148e8034e2d7/#diff-1) (current release/2016); It would be great if you helped us by doing the tests :)

Here some tips about  about how to [configure minidlna](https://wiki.archlinux.org/index.php/ReadyMedia)

**Comment by SteveEbey73742** [comment]
Single user, running fedora 25, cinnamon desktop on AMD FX-6300 at 3.9Ghz.  40+ years in the field of technology, from hex keypad 6502 assembly code, to C++ medical research investigation programming.  Owner of a computer business, that specializes in network setup, and penetration testing.  Since my nic is using a static ip, and I am the only user on the computer, i see no need to force it to try and find the named interface.  Having said that, I do think it would be nice, for others that need the feature, to make it work.  I will try 1.1.6 and see if it finds the listening port, or has to be restarted, after login, to work properly.

**Comment by kuboosoft** [comment]
@SteveEbey73742 @paulcarroty @sergiomb2  Updated now; Now you can test it :)

**Comment by SteveEbey73742** [comment]
Downloaded and installed, same config as 1.1.5 and it crashed.  shown below is output of systemctl status minidlna. Hope this helps.

```
● minidlna.service - MiniDLNA is a DLNA/UPnP-AV server software
   Loaded: loaded (/usr/lib/systemd/system/minidlna.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Sun 2017-04-23 07:34:11 CDT; 18s ago
 Main PID: 5945 (code=exited, status=203/EXEC)

Apr 23 07:34:11 linuxbox-desktop systemd[1]: Started MiniDLNA is a DLNA/UPnP-AV server software.
Apr 23 07:34:11 linuxbox-desktop systemd[1]: minidlna.service: Main process exited, code=exited, status=203/EXEC
Apr 23 07:34:11 linuxbox-desktop systemd[1]: minidlna.service: Unit entered failed state.
Apr 23 07:34:11 linuxbox-desktop systemd[1]: minidlna.service: Failed with result 'exit-code'.
```
and here is output of 1.1.5 
```
minidlna.service - MiniDLNA is a DLNA/UPnP-AV server software
   Loaded: loaded (/usr/lib/systemd/system/minidlna.service; enabled; vendor preset: disabled)
   Active: active (running) since Sun 2017-04-23 07:36:16 CDT; 7s ago
 Main PID: 6429 (minidlnad)
    Tasks: 2 (limit: 4915)
   CGroup: /system.slice/minidlna.service
           └─6429 /usr/sbin/minidlnad -S

Apr 23 07:36:16 linuxbox-desktop systemd[1]: Started MiniDLNA is a DLNA/UPnP-AV server software.
Apr 23 07:36:16 linuxbox-desktop minidlnad[6429]: minidlna.c:1030: warn: Starting MiniDLNA version 1.1.5.
Apr 23 07:36:16 linuxbox-desktop minidlnad[6429]: minidlna.c:1070: warn: HTTP listening on port 8200
Apr 23 07:36:16 linuxbox-desktop minidlnad[6429]: minidlna.c:1075: warn: TiVo support is enabled.
Apr 23 07:36:16 linuxbox-desktop minidlnad[6429]: inotify.c:198: warn: WARNING: Inotify max_user_watches [8192] is low or close to the number of used watches [2] and I do not have permission
```

i will stay with 1.1.5 for now, since it works, as long as interface name is commented.

**Comment by SteveEbey73742** [comment]
Further testing, the errors i got were from the system service file, so i reverted to using my original service file, the problem with network interface is still there.

● minidlna.service - MiniDLNA is a DLNA/UPnP-AV server software
   Loaded: loaded (/usr/lib/systemd/system/minidlna.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2017-04-25 19:22:08 CDT; 1min 16s ago
 Main PID: 1128 (minidlnad)
    Tasks: 2 (limit: 4915)
   CGroup: /system.slice/minidlna.service
           └─1128 /usr/sbin/minidlnad -S

Apr 25 19:22:08 linuxbox-desktop systemd[1]: Started MiniDLNA is a DLNA/UPnP-AV server software.
Apr 25 19:22:15 linuxbox-desktop minidlnad[1128]: minidlna.c:1022: warn: Starting MiniDLNA version 1.1.6.
Apr 25 19:22:15 linuxbox-desktop minidlnad[1128]: minidlna.c:1062: warn: HTTP listening on port 8200
Apr 25 19:22:15 linuxbox-desktop minidlnad[1128]: minidlna.c:1067: warn: TiVo support is enabled.
Apr 25 19:22:15 linuxbox-desktop minidlnad[1128]: getifaddr.c:109: error: Network interface enp2s0 not found
Apr 25 19:22:15 linuxbox-desktop minidlnad[1128]: inotify.c:199: warn: WARNING: Inotify max_user_watches [8192] is low or close to the number of used watches [2] and I do not have permi

**Comment by kuboosoft** [comment]
mmmm, then is a problem in the source code... maybe is necessary open a [ticket](https://sourceforge.net/p/minidlna/bugs/?source=navbar) ; why the developer no open a github repository for it?... It can open the collaboration... a [similar issue](https://sourceforge.net/p/minidlna/bugs/297/)

**Comment by SteveEbey73742** [comment]
ticket opened.

**Comment by sergiomb2** [comment]
rpmfusion ticket closed , it means minidlna was been built with new systemd units , thanks  , should be available in rpmfusion repos in 2 or 3 days.
